jenkins-apb/roles/provision-jenkins-apb/tasks/main.yml

```diff
@@ -0,0 +1,178 @@
+# ignore_errors is a work around.
+#
+# NOTE: In annotations to_json is a workaround.  There is an
+# issue with annotations that contain JSON string.
+# ----------------------------
+
+- name: Create ServiceAccount
+  k8s_v1_service_account:
+    name: "{{ jenkins_service_name }}"
+    namespace: "{{ namespace }}"
+    annotations:
+      serviceaccounts.openshift.io/oauth-redirectreference.jenkins: "{{ oauth_redirect_reference_jenkins | to_json }}"
+    state: "{{ state }}"
+  ignore_errors: True
+
+- name: Create RoleBinding from Ansible Template
+  template:
+    src: rolebinding.yaml.j2
+    dest: /tmp/rolebinding.yaml
+  register: rolebinding
+
+# Create RoleBinding
+# ----------------------------
+# NOTE: The openshift_v1_role_binding module does not function correctly.
+# Using the src it cannot determine the correct parameter to use
+# for roleRef.name
+#
+# Using the module parameters the role_ref_name contains the name of the object
+# which is incorrect.
+#
+# Workaround: use the command module and oc. This will error if ran more than once
+# so ignore_errors
+# ----------------------------
+
+- name: Create RoleBinding
+  command: "oc create -f {{ rolebinding.dest | default(rolebinding.path) }} -n {{ namespace }}"
+  ignore_errors: True
+
+# ----------------------------
+# Create S2I build of Jenkins
+# ----------------------------
+# If source_git_uri is defined let's create an ImageStream and BuildConfig
+# so that we can create a customized version of Jenkins.
+# ----------------------------
+
+- name: Create ImageStream
+  openshift_v1_image_stream:
+    name: "{{ jenkins_service_name }}"
+    namespace: "{{ namespace }}"
+    state: "{{ state }}"
+  when: source_git_uri is defined and
+        source_git_ref is defined and
+        source_context_dir is defined
+
+- name: Create BuildConfig
+  openshift_v1_build_config:
+    name: "{{ jenkins_service_name }}"
+    namespace: "{{ namespace }}"
+    spec_output_to_kind: "ImageStreamTag"
+    spec_output_to_name: "jenkins:latest"
+    spec_source_context_dir: "{{ source_context_dir }}"
+    spec_source_git_uri: "{{ source_git_uri }}"
+    spec_source_git_ref: "{{ source_git_ref }}"
+    spec_source_type: "Git"
+    spec_strategy_source_strategy__from_kind: "ImageStreamTag"
+    spec_strategy_source_strategy__from_name: "jenkins:2"
+    spec_strategy_source_strategy__from_namespace: "openshift"
+    spec_triggers:
+      - type: ConfigChange
+      - type: ImageChange
+    state: "{{ state }}"
+  when: source_git_uri is defined and
+        source_git_ref is defined and
+        source_context_dir is defined
+
+# ----------------------------
+# DeploymentConfig
+# ----------------------------
+# An ansible template is used to support the ability for
+# this APB to deploy a persistent or ephemeral Jenkins instance.
+# ----------------------------
+- name: Create DeploymentConfig from Template
+  template:
```

Comment:
Interesting approach. Other APBs `include` different files depending on the vars; this looks more clean